### ISSUE TITLE
Create analogue AI for solo/coop missions

### ISSUE THREAD

**Issue by Wollivan** [Wollivan/SquadronLeader#10]
For there to be coop campaigns against non human players, there needs to be a systme similar to Heroes of the Aturi Cluster for controlling enemy ships.

**Comment by Wollivan** [comment]
Please refer to this discussion:

https://github.com/Wollivan/SquadronLeader/discussions/31

**Comment by Wollivan** [comment]
A Few feature ideas I've had, again based on the amazing work already done by the HotAC team.
We have the distinct advantage of having a simpler rules base that the AI needs to fit into, so the AI _should_ be simpler... _should_!

The main aim is to get the enemy ship to a point where they can fire upon the players.

So we need to figure out if the aim is purely to be facing the player and go towards them, or is it also to account for the player moving second and giving the AI as much a chance as possible to have the player in their sights?

In HotAC, the movement is based upon the the range of the nearest enemy, and the "quadrant" they sit in. This could be an option with the changes proposed in this pull request:
https://github.com/Wollivan/SquadronLeader/pull/35

In HotAC the AI also cares about if the ships are moving closer or further apart.

So the main questions to aid the next steps are:
- Should the AI have different moves for if they are moving first or second
    - i.e. if they are going first, just to face the player, if they are going second, get as close to the player as possible
- How many possible moves should the AI have per quadrant, and per range.
    - Should it involve a die roll or just be 1 move per quadrant and per range (so 3 per quadrant)
- Should the AI prioritise ships in front of it, if there is one, before going for ships behind it
- Should there be a difference between when the AI is pursuing vs being pursued
- Should the AI have altitude and follow altitude rules

I have made a VERY basic mockup of a potential printable AI template that you would hold over the AI token for their move for people to have a look at and judge, here is an image of it:

<img width="523" alt="Screenshot 2022-05-13 at 00 47 31" src="https://user-images.githubusercontent.com/91621088/168185505-31a63b4c-7d5d-4fcd-ab3f-9df5701eb51d.png">
<img width="766" alt="Screenshot 2022-05-13 at 00 52 04" src="https://user-images.githubusercontent.com/91621088/168185511-0d5d81e1-9930-4e6b-9598-f4f6a6ef67b9.png">


This is based on the following assumptions:
- You hold the template over the AI to determine the move based on the closest human player
- There is one move per range/per quadrant
- The AI has unlimited speed 
- The AI doesn't require altitude to perfom a Split S maneuver

**Comment by Wollivan** [comment]
Calling @mevdev @penyuan @Winehorn for their opinion :)

**Comment by lifelike** [comment]
I think the "words" look almost nicely aligned, radiating out from the center, but are slightly off, in particular the lone T and P to the right in the top image could  be moved slightly to the left. Maybe try to align all text along lines from the center going through the center of each arc? I know it has no functional effect, but I think it would make it look better.

**Comment by Wollivan** [comment]
> I think the "words" look almost nicely aligned, radiating out from the center, but are slightly off, in particular the lone T and P to the right in the top image could  be moved slightly to the left. Maybe try to align all text along lines from the center going through the center of each arc? I know it has no functional effect, but I think it would make it look better.

I completely agree! This is a very rough go at the model, I just wanted to throw the idea out to everyone.

I've been working on it today, I'm not on my my laptop right now so can't post a pic, but I'm working on making the whole thing a lot more mathematical and straight.

Thanks lifelike! :)

**Comment by penyuan** [comment]
Rules wise, I'd suggest starting with a simpler set of AI rules that at least works, then add complexity later, though I think it makes sense for the AI to have altitude-related behavior!

**Comment by Wollivan** [comment]
So I had a long discussion with a friend this weekend, and he made a very helpful suggestion that I hadn't considered.


The printable part for the AI only has to be very simple, if not no existent.
Treat the AI like a D&D look up table, based on range and angle, no fancy template with rules written on it required.


So I'm going to work on a simple AI and rework the website/repo so we can have multiple pages. Then I can create an expansion page with the start of the first campaign that people can add to.

Exciting times!
